### ISSUE TITLE
[notification] investigate deprecating actions from toast

### ISSUE THREAD

**Issue by aagonzales** [carbon-design-system/carbon#6433]
### Summary

Actions inside of toast notifications have been found to not be WCAG 2.1 compliant.

We need to clearly document why why they are not accessible and if there are any acceptable solutions (ie notifications collect and available in accessible area).

If we do not want to support any scenario that includes an action in toast then we need to full deprecate "action" in toast notification. (V11 or sooner?)

If we want accept the available in accessible area solution then we need to clear document this as the only why to have an action. 

### References

- The following is the passage on the notification [usage](https://carbondesignsystem.com/components/notification/usage#toast-notifications) page that needs clarification and decision. 

> **Action**
> Carbon toast notifications are not fully accessible when they include interactive content. Using links and other interactive content in toasts breaks WCAG 2.1 compliance and should therefore be avoided. The Carbon team is aware of the accessibility concerns and is currently working on deprecating this functionality.

- This is a slack thread where it comes up. Kyle Richard says there team heavily realizes on this method.
https://ibm-studios.slack.com/archives/C0M053VPT/p1594251900120100

**Comment by aagonzales** [comment]
Example of persistent toast 
![image](https://user-images.githubusercontent.com/11670886/87588813-e4494080-c6a9-11ea-8fca-8b1c9bf06592.png)

**Comment by aagonzales** [comment]
Questions to answer. 
- need to build a non-modal dialog notification? 
- if so does it still need to live in a drawer somewhere?

**Comment by vpicone** [comment]
We can just disable the timeout dismissal when actions are provided. I talk about it in my issue here https://github.com/carbon-design-system/carbon/issues/5055

**Comment by tombrunet** [comment]
To add our +1: We were confused by the deprecation note. We're assuming this is due to the guidance to automatically dismiss. The React component leaves that under the control of the developer and does not automatically dismiss as-is. Is there something else causing issues?

**Comment by dakahn** [comment]
closing this out since we've moved forward with notification/toast v11 refactor and guidance 👍🏽
